lib/logstash/filters/jdbc_static.rb

```diff
@@ -0,0 +1,256 @@
+  config :tag_on_failure, :validate => :array, :default => ["_jdbcstreamingfailure"]
+
+  # Append values to the `tags` field if no record was found and default values were used
+  config :tag_on_default_use, :validate => :array, :default => ["_jdbcstreamingdefaultsused"]
+
+  # Remote Load DB JDBC driver library path to third party driver library.
+  config :jdbc_driver_library, :validate => :path
+
+  # Remote Load DB JDBC driver class to load, for example "oracle.jdbc.OracleDriver" or "org.apache.derby.jdbc.ClientDriver"
+  config :jdbc_driver_class, :validate => :string, :required => true
+
+  # Remote Load DB JDBC connection string
+  config :jdbc_connection_string, :validate => :string, :required => true
+
+  # Remote Load DB JDBC user
+  config :jdbc_user, :validate => :string
+
+  # Remote Load DB JDBC password
+  config :jdbc_password, :validate => :password
+
+  # Local Lookup DB JDBC driver class to load, for example "org.apache.derby.jdbc.ClientDriver"
+  config :lookup_jdbc_driver_class, :validate => :string, :required => false, :default => "org.apache.derby.jdbc.EmbeddedDriver"
+
+  # Local Lookup DB JDBC driver library path to third party driver library.
+  config :lookup_jdbc_driver_library, :validate => :path, :required => false
+
+  # Local Lookup DB JDBC connection string
+  config :lookup_jdbc_connection_string, :validate => :string, :required => false, :default => "jdbc:derby:memory:localdb;create=true"
+
+  class << self
+    alias_method :old_validate_value, :validate_value
+
+    def validate_value(value, validator)
+
+      result = value
+      if validator.is_a?(Array) && validator.first.is_a?(Class)
+        validation_error = validator.first.validate(value)
+        unless validation_error.nil?
+          return false, validation_error
+        end
+      else
+        return old_validate_value(value, validator)
+      end
+      [true, result]
+    end
+  end
+
+  public
+
+  def register
+    # @symbol_parameters = @parameters.inject({}) {|hash,(k,v)| hash[k.to_sym] = v ; hash }
+    prepare_runner
+    @loader_runner.initial_load
+  end
+
+  def filter(event)
+    @processor.enhance(event)
+    filter_matched(event)
+  end
+
+  # ---------------------
+  # temp while developing
+  def local
+    @processor.local
+  end
+  # ---------------------
+
+
+  def stop
+    @scheduler.stop if @scheduler
+    @parsed_loaders.each(&:close)
+    @loader_runner.close
+  end
+
+  private
+
+  def prepare_runner
+    @parsed_loaders = @loaders.map do |options|
+      add_plugin_configs(options)
+      loader = Util::Loader.new(options)
+      loader.build_remote_db
+      loader
+    end
+
+    @processor = Util::LookupProcessor.new(@lookups, global_lookup_options(), @logger)
+
+    if @schedule
+      require "rufus/scheduler"
+      @loader_runner = Util::RepeatingLoadRunner.new(
+          @processor.local, @parsed_loaders, @local_db_objects)
+
+      @scheduler = Rufus::Scheduler.new(:max_work_threads => 1)
```

Comment:
`overlap` is a Job option not a Scheduler.new option and it does not pass down to the cron job.